xdsl/interactive/app.py

```diff
@@ -130,6 +132,11 @@ def action_quit_app(self) -> None:
         """An action to quit the app."""
         self.exit()
 
+    @on(Button.Pressed, "#clear_input_button")
+    def on_clear_input_button_pressed(self, event: Button.Pressed) -> None:
+        """When the "Clear Input" button is pressed, the input IR TextArea is cleared and the current_module is updated"""
```

Comment:
```suggestion
```

I think documentation is excessive here, if not inaccurate.
First, the `@on` decorator makes it clear which event and widget are handled here; no need to comment IMHO.
Second, "and the current_module is updated." part is wrong; that is done by `update_current_module`, right?

Reply:
Oh BTW I don't think the `@on` decorator is needed, the name of the function should already be enough to specify the behaviour

Reply:
Ah no it would have been if the class name was custom, but it's not in our case. In that case, I'd recommend renaming the function `clear_input` to make the role of the function clearer, and keep the `@on` decorator.